src/ring_theory/dedekind_domain.lean

```diff
@@ -592,4 +593,9 @@ noncomputable instance fractional_ideal.comm_group_with_zero
   mul_inv_cancel := λ I, fractional_ideal.mul_inv_cancel,
   .. fractional_ideal.comm_semiring }
 
+noncomputable instance ideal.comm_cancel_monoid_with_zero
+  [is_dedekind_domain A] : comm_cancel_monoid_with_zero (ideal A) :=
```

Comment:
Don't you promise a `comm_group_with_zero` in the comments above?

Reply:
*Fractional* ideals are a group with zero, *integral* ideals are just a monoid. I'll clarify the docstring a bit.